File: src/main/java/com/tunisij/financeplanner/salary/Salary.java
package com.tunisij.financeplanner.salary;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "salary")
public class Salary {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "base_salary")
    private Double baseSalary;

    @Column(name = "annual_bonus")
    private Double bonus;

    @Column(name = "retirement_match")
    private Double retirementMatch;
}

File: src/main/java/com/tunisij/finance-planner-ui/src/app/objects/salary.ts
export class Salary {
  id?: String;
  name: String;
  baseSalary = 0;
  bonus = 0;
  retirementMatch = 0;

  constructor(name: String, baseSalary: number, bonus: number, retirementMatch: number, id?: String) {
    this.name = name;
    this.baseSalary = baseSalary;
    this.bonus = bonus;
    this.retirementMatch = retirementMatch;
    this.id = id;
  }
}
